### PR TITLE
issue352 Point_of_order-feature fromOS3 implemented including extende…

### LINKED ISSUE #352: Implement the "point of order" feature from OS3
(Documentation follows...)

### PR DESCRIPTION
**Pull request by r-peschke**
issue352 Point_of_order-feature from OS3 implemented including extended tests
OS3 features from server/openslides/agenda/models.py, class SpeakerManager

closes #352

### CODE REVIEW

File: openslides_backend/action/actions/speaker/create_update_delete.py [hunk 1]
```diff
@@ -19,26 +19,74 @@ class SpeakerCreateAction(CreateActionWithInferredMeeting):
...
             filter=filter_obj,
-            mapped_fields=["user_id"],
+            mapped_fields=["user_id", "point_of_order"],
             lock_result=True,
         )
         for speaker in speakers.values():
-            if speaker["user_id"] == instance["user_id"]:
+            if speaker["user_id"] == instance["user_id"] and speaker.get(
+                "point_of_order"
+            ) == instance.get("point_of_order"):
```
Comment (FinnStutzenstein) [inline openslides_backend/action/actions/speaker/create_update_delete.py:89]: `instance.get("point_of_order")`may be `None`. Maybe define a `point_of_order` variable with either `True` or `False`.

File: tests/system/action/speaker/test_create_point_of_order.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+            ],
+        )
+        self.assert_status_code(response, 200)
+        self.assert_model_exists(
+            "speaker/3",
+            {"user_id": 9, "list_of_speakers_id": 23, "point_of_order": True},
+        )
+        self.assert_model_exists(
+            "speaker/4",
+            {"user_id": 9, "list_of_speakers_id": 23, "point_of_order": None},
```
Comment (FinnStutzenstein) [inline tests/system/action/speaker/test_create_point_of_order.py:116]: Maybe just leave the key `point_of_order` out.

**Comment by FinnStutzenstein** [comment]
The logic looks good so far!

File: openslides_backend/action/actions/speaker/create_update_delete.py (hunk 1 above)
Comment (r-peschke) [reply openslides_backend/action/actions/speaker/create_update_delete.py:89]: Done

File: tests/system/action/speaker/test_create_point_of_order.py (hunk 2 above)
Comment (r-peschke) [reply tests/system/action/speaker/test_create_point_of_order.py:116]: Done

File: openslides_backend/services/datastore/adapter.py [hunk 3]
```diff
@@ -263,6 +268,10 @@ def min(
             f"Start MIN request to datastore with the following data: {command.data}"
         )
         response = self.retrieve(command)
+        if lock_result:
+            self.update_locked_fields(
+                CollectionField(collection, field), response.get("position")
+            )
         return response
```
Comment (jsangmeister) [inline openslides_backend/services/datastore/adapter.py:275]: We should probably adjust the adapter (for this and the other aggregate functions) to just return the min and not the position. If you want, you can do this here or open another PR for it.

**Review by jsangmeister — APPROVED** [review]

**Comment by FinnStutzenstein** [comment]
@r-peschke Please add this PR to the OpenSlides 4 project, so it appears in the board.

File: openslides_backend/services/datastore/adapter.py (hunk 3 above)
Comment (r-peschke) [reply openslides_backend/services/datastore/adapter.py]: Done

File: openslides_backend/services/datastore/interface.py [hunk 4]
```diff
@@ -12,7 +12,7 @@
 PartialModel = Dict[str, Any]
 Found = TypedDict("Found", {"exists": bool})
 Count = TypedDict("Count", {"count": int})
-Aggregate = Dict[str, Any]  # TODO: This interface seams to be wrong.
+OptionalInt = Optional[int]
```
Comment (jsangmeister) [inline openslides_backend/services/datastore/interface.py:15]: This is just confusing ^^ please just write the full type where you use it.

**Review by jsangmeister — CHANGES_REQUESTED** [review]
Please also remove the position for the other aggregate functions (`exists` and `count`). Also for the future, you don't have to adapt everything in the `old` directory to new changes - that's why it's called old :D Btw, @normanjaeckel, do you still need these files or can they finally be removed? They are not used in any way in the backend as-is, so if you are not using them in your local workflow, we can safely delete them.

**Comment by FinnStutzenstein** [comment]
> do you still need these files or can they finally be removed?

We had a phona call about them last year, if I remember correctly. We said that these files can be removed, if there are conflics/problems with them.

**Review by FinnStutzenstein — APPROVED** [review]
Just some wordings in comments

File: openslides_backend/services/datastore/interface.py (hunk 4 above)
Comment (r-peschke) [reply openslides_backend/services/datastore/interface.py]: Done

**Comment by r-peschke** [comment]
> > do you still need these files or can they finally be removed?
> 
> We had a phone call about them last year, if I remember correctly. We said that these files can be removed, if there are conflics/problems with them.

Removed the whole directory

File: openslides_backend/action/actions/speaker/create_update_delete.py [hunk 5]
```diff
@@ -19,26 +19,74 @@ class SpeakerCreateAction(CreateActionWithInferredMeeting):
...
+            instance["weight"] = -1 if weight is None else weight - 1
+        return instance
+
     def validate_fields(self, instance: Dict[str, Any]) -> Dict[str, Any]:
         """
-        Checks that a new speaker does not already exist on the list of speaker as
-        comming speaker (with begin_time == None)
+        Checks
+        - that a new speaker does not already exist on the list of speaker as
+        comming speaker (with begin_time == None), but allows one additional with point_of_order speaker per user
```
[the remark this answers is not in the document]
Comment (r-peschke) [reply openslides_backend/action/actions/speaker/create_update_delete.py:44]: Done

File: openslides_backend/action/actions/speaker/create_update_delete.py [hunk 6]
```diff
@@ -19,26 +19,74 @@ class SpeakerCreateAction(CreateActionWithInferredMeeting):
...
+        - checks, if points_of_order are used inthis meeting
```
[the remark this answers is not in the document]
Comment (r-peschke) [reply openslides_backend/action/actions/speaker/create_update_delete.py:45]: Done

File: tests/unit/test_action_handler.py [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+from openslides_backend.action.util.typing import Payload
+
+"""
+ACHTUNG !!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+Angefangen zum PullRequest #424 Adjust WR to per-action-base with locked fields
+Joshau sieht in als Unittest, ist aber eher ein system-Test, da er
+mindestens den Datastore-Service braucht, um seine WriteRequests zu erzeugen
+
+"""
```
Comment (jsangmeister) [inline tests/unit/test_action_handler.py:14]: Why is this in this PR? Also, please write comments always in english to keep it uniform.

**Review by jsangmeister — CHANGES_REQUESTED** [review]
Also, please do a search for `/?old/?` - there is at least one exception for it in the Makefile (maybe more, maybe also somewhere else) that can be removed now.

File: tests/unit/test_action_handler.py (hunk 7 above)
Comment (r-peschke) [reply tests/unit/test_action_handler.py:14]: > Why is this in this PR? Also, please write comments always in english to keep it uniform.
Sorry, this has been a remark for myself to not commit this file anywhere ;-) => deleted

**Comment by r-peschke** [comment]
> Also, please do a search for `/?old/?` - there is at least one exception for it in the Makefile (maybe more, maybe also somewhere else) that can be removed now.

removed from several config-files
